doc/deployment.md

```diff
@@ -0,0 +1,134 @@
+
+Technically these clients are all the same, and a single executable may fulfill several or all of these roles at the same time. Conceptually it makes sense to do this differentiation between different intents of interacting with the VSS tree.
+
+Intuitively, it can be seen that the security and safety requirements in an end-2-end system differ for these roles.
+
+
+# Deployment Blueprint 1: Internal API
+
+You are using VSS and KUKSA.val as an internal API in your system to ease system development. You control the whole system/system integration. 
+
+![Deployment Blueprint 1: Internal API](./pictures/deployment_blueprint1.svg)
+
+| Aspect           | Design Choice |
+| ---------------- | ------------- |
+| Users            | You           | 
+| System Updates   | Complete      | 
+| Security         | None/Fixed    | 
+| VSS model        | Static        | 
+| KUKSA deployment | Firmware      | 
+
+You are not exposing any VSS API to external parties, you control all components interacting with VSS, the system is state/composition is under your control. In this case you make KUKSA APIs available only within your system. You might even opt to disabling encryption for higher performance and not using any tokens for authentication. We would still recommend leaving basic security measures intact, but this deployment does not need fine-grained control of permission rights or fast rotation/revocation of tokens.
+
+# Deployment Blueprint 2: Exposing a subset of VSS to another system
+
+You control the system (e.g. Vehicle Computer), that has KUKSA.val deployed. You want to make a subset of capabilities available to another system or middleware, that may have its own API/security mechanisms. An example would be an Android Automotive based IVI system, 
+
+![Deployment Blueprint 2: Exposing a subset of VSS to another system](./pictures/deployment_blueprint2.svg)
+
+| Aspect            | Design Choice |
+| ----------------- | ------------- |
+| Users             | Other trusted platforms             |
+| System Updates    | Firmware on controlled system, unknown on third party system      | 
+| Security          | TLS+Foreign platform token    | 
+| VSS model         | Static        | 
+| KUKSA deployment | Firmware or Software package       | 
+
+In this deployment the foreign system is treated as a single client, which we has a certain level of access and trust. Whether that system restricts access further for certain hosted apps is opaque to KUKSA.val. In this deployment you need to enable and configure TLS to provide confidentiality, as well as providing a single token limiting the access of the foreign platform.
+
+If the token is time limited, the foreign platform needs to be provided with a new token in time, in case certificates in the KUKSA.val side are changed, the foreign system needs to be updated accordingly. 
```

Comment:
```suggestion
If the token is time limited, the foreign platform needs to be provided with a new token in time. In case certificates on the KUKSA.val side are changed, the foreign system needs to be updated accordingly. 
```